### ISSUE TITLE
Bug / I don't know what I do: When running sudo debpack 1.0

### ISSUE THREAD

**Issue by NeonCrafter13** [zmarffy/debpacker#1]
When I run this command: `sudo debpack 1.0` I get an Error:
_
2020-10-30 18:54:35,108 [CRITICAL] 
Traceback (most recent call last):
  File "/usr/local/bin/debpack", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/debpacker/__main__.py", line 201, in main
    "default": "{} <{}>".format(_run_command("git config user.name"), _run_command("git config user.email")),
  File "/usr/local/lib/python3.8/dist-packages/debpacker/__main__.py", line 101, in _run_command
    raise e
  File "/usr/local/lib/python3.8/dist-packages/debpacker/__main__.py", line 98, in _run_command
    return check_output(command, stderr=PIPE, shell=shell).decode().strip()
  File "/usr/lib/python3.8/subprocess.py", line 411, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/usr/lib/python3.8/subprocess.py", line 512, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command 'git config user.name' returned non-zero exit status 1.
_

Doe's anyone know how I can fix this in case I'm doing something wrong.

Because when i run:
`git config user.name`

I get:
`NeonCrafter13`

I hope that you can help me. Btw my OS is Linux Mint.

**Comment by zmarffy** [comment]
Hi! Thank you for taking the time to check out this project.

You are experiencing this because you ran it with `sudo`. Running it with `sudo` will run it as root, and I'm sure your root user does not have git creds set up.

Just don't run it with `sudo` and you should be good!

**Comment by NeonCrafter13** [comment]
Thanks that was my problem.
But I have a different question now:
Do I need to specify all files under build or only the "main" file.

Also your Documentation could be expanded.

**Comment by zmarffy** [comment]
Sure thing. This is for sure something to add to the docs

It depends what you're trying to pack. The `build` script is an executable that contains your code to compile your source files. What kind of project are you trying to pack? Does it require any compilation? This certainly needs docs; I plan to write them next week.

Let's say you wrote a hello world program in C++ called `helloworld.cpp`. Its code is as follows.

```c++
// Your First C++ Program

#include <iostream>

int main() {
    std::cout << "Hello World!";
    return 0;
}
```

If it is in the root of the source code folder, the `.debpack/build` script would be a one-liner.

```bash
g++ helloworld.cpp -o helloworld
```

This will run to compile your code and create `helloworld` in the source directory, an actual executable.

The `.debpack/config.json` file would have one thing under the `files` section under `build`. It would be `"helloworld" : "/usr/bin/helloworld"`. The key is what the file in the source folder is (compiled or not, and, in this case, it is compiled). The value is where it will go when it is installed on a computer.

In conclusion, you need to specify whatever files are necessary to move to certain parts of your filesystem under `build` in the config file, and you need to write your compilation process in the `.debpack/build` script.

Does that make sense?

**Comment by NeonCrafter13** [comment]
it makes seens but how do you do that in Python. What do you do if you have multiple files and import them could you give an example of that?
You're explanation makes sens but i am also confused.

Thats the project i am trying to get into a .deb:
https://github.com/NeonCrafter13/Homeworkplanner

I Hope I am not bothering you to much.

**Comment by zmarffy** [comment]
You're not bothering me too much! I'm always here to help.

What files you want to drop off where is your choice. For something like Python, though, they need to be in a place where they can import each other.

I would actually recommend looking into [Python packaging](https://the-hitchhikers-guide-to-packaging.readthedocs.io/en/latest/creation.html) eventually because this is the arguably most "correct" and "clean" way to go about distributing Python packages. DEB files may not be the best.

However, I still am very happy you are using my tool, so I'll give you pointers on how to go about that so you can get it to work!

[Your `main.py` file imports your `task.py` file.](https://github.com/NeonCrafter13/Homeworkplanner/blob/main/main.py#L23) That means `task.py` must be in a place that Python knows to import it from or you must direct the code to import it directly somehow. There are four ways to go about this. One, having it in its own package. Two, having it in the same folder that `main.py` is in. Three, appending to `sys.path` in Python before you import it. We'll save the fourth for last because I think it's the cleanest.

So. Number one. Having `task.py` in its own package is a little weird because if you're gonna do that, you may as well just package your whole program as a PIP package, not just `task.py`. (`debpack` does not create PIP packages; that is a different beast entirely, but I am glad to help with that somewhere else if you want to attempt it.) Number two. Having `task.py` in the same folder as `main.py` is a little weird because I would assume you want them placed in `/usr/bin/` (so that they can be ran very easily via the command line), and that would make `task` then a command on your computer that invokes `task.py` itself, which I'm sure you do not want to do, as `task.py` probably doesn't even do anything by itself, right? And number three. Three is a bit hacky. That's all there is to say about that.

There is a fifth option, combining all your code into one file, but that looks ugly of course and becomes a little difficult to maintain.

So let's talk about the fourth option that seems the most correct to me. You could make the `config.json` file for `debpack` place the helper modules into a sectioned off folder and `main.py` in `/usr/bin/`. And in `main.py`, you could write [the code for importing directly from that folder](https://stackoverflow.com/a/67692/6131504).

I'll submit a PR to your repo that does just that. Pay attention to your repo and you'll see my code once I finish it. 😄

**Comment by NeonCrafter13** [comment]
Thanks, that hole topic of packaging is way to confussing for me.

On Windows I just create Executables using https://pypi.org/project/auto-py-to-exe/ which has a graphical interface.
I really apreceate that you take so much time.

**Comment by zmarffy** [comment]
I understand! It was certainly difficult to wrap my head around it. I had to take multiple hours over multiple days.

I should be done with the PR very soon.

**Comment by NeonCrafter13** [comment]
Thanks for helping me out.

**Comment by zmarffy** [comment]
Sure thing. Hopefully my techniques make sense to you.

More actual docs to come this week.

**Comment by zmarffy** [comment]
I updated [the docs](https://github.com/zmarffy/debpacker/blob/master/README.md) if you would like more clarification.
